Wire gradle plugin can error out when gradle is running in debug mode

I have a project where I'm basically running `gradle :foo:bar` and it was producing an error. To debug it further I needed to run gradle in debug mode, so I ran `gradle --debug :foo:bar`. However, when running under debug mode, I got a *different* error, coming from Wire. It seems that Wire has behaviour differences between debug and info modes, and can result in errors in debug mode that would get skipped in info mode.

The error produced in debug mode looks like this:

```
Caused by: org.gradle.api.internal.provider.MissingValueException: Cannot query the value of task ':client:generateMainProtos' property 'protoLibraryOutput' because it has no value available.
```

and the relevant part of the stack trace looks like this:
<img width="1139" alt="Screenshot 2023-09-22 at 1 24 21 PM" src="https://github.com/square/wire/assets/485789/44f20cfa-2901-4f8c-9a29-b78637d23b27">

(Sorry for screenshot, but copying out of a build scan doesn't seem to work)

--- a/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireInput.kt
+++ b/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireInput.kt
@@ -28,7 +28,6 @@ import org.gradle.api.artifacts.FileCollectionDependency
 import org.gradle.api.file.SourceDirectorySet
 import org.gradle.api.internal.catalog.DelegatingProjectDependency
 import org.gradle.api.internal.file.FileOrUriNotationConverter
-import org.gradle.api.logging.Logger
 import org.gradle.api.provider.Provider
 
 /**
@@ -141,16 +140,6 @@ internal class WireInput(var configuration: Configuration) {
     } catch (e: Exception) {
       false
     }
-
-  fun debug(logger: Logger) {
-    configuration.dependencies.forEach { dep ->
-      val srcDirs = ((dep as? FileCollectionDependency)?.files as? SourceDirectorySet)?.srcDirs
-      val includes = dependencyFilters[dep] ?: listOf()
-      logger.debug("dep: $dep -> $srcDirs")
-      logger.debug("$name.files for dep: ${configuration.files(dep)}")
-      logger.debug("$name.includes for dep: $includes")
-    }
-  }
 
   fun toLocations(project: Project): Provider<List<InputLocation>> {
     // We create a provider to support lazily created locations which do not exist at

--- a/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt
+++ b/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt
@@ -220,10 +220,6 @@ class WirePlugin : Plugin<Project> {
         task.description = "Generate protobuf implementation for ${source.name}"
         task.source(protoSourceInput.configuration)
 
-        if (task.logger.isDebugEnabled) {
-          protoSourceInput.debug(task.logger)
-          protoPathInput.debug(task.logger)
-        }
         val outputDirectories: List<String> = buildList {
           addAll(
             targets

--- a/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireTask.kt
+++ b/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireTask.kt
@@ -148,13 +148,6 @@ abstract class WireTask @Inject constructor(objects: ObjectFactory) : SourceTask
     }
     if (includes.isEmpty() && excludes.isEmpty()) logger.info("NO INCLUDES OR EXCLUDES")
 
-    if (logger.isDebugEnabled) {
-      logger.debug("roots: ${roots.orNull}")
-      logger.debug("prunes: ${prunes.orNull}")
-      logger.debug("rules: ${rules.orNull}")
-      logger.debug("targets: ${targets.orNull}")
-    }
-
     inputFiles.forEach { fileObj ->
       check(fileObj.exists()) {
         "Invalid path string: \"${fileObj.path}\". Path does not exist."

